ReactWindows/ReactNative.Shared/Bridge/Queue/ReactQueueConfigurationFactory.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Reactive.Concurrency;
+#if WINDOWS_UWP
+using Windows.ApplicationModel.Core;
+using Windows.UI.Core;
+#else
+using System.Windows;
+using System.Windows.Threading;
+#endif
+
+namespace ReactNative.Bridge.Queue
+{
+    /// <summary>
+    /// Factory to create the React queue configuration.
+    /// </summary>
+    class ReactQueueConfigurationFactory
+    {
+        private readonly IActionQueueFactory _dispatcherQueueFactory;
+        private readonly IActionQueueFactory _javaScriptQueueFactory;
+        private readonly IActionQueueFactory _nativeModulesQueueFactory;
+
+        /// <summary>
+        /// Instantiates the <see cref="ReactQueueConfigurationFactory"/>.
+        /// </summary>
+        /// <param name="dispatcherQueueFactory">The dispatcher queue factory.</param>
+        /// <param name="javaScriptQueueFactory">The JavaScript queue factory.</param>
+        /// <param name="nativeModulesQueueFactory">The native modules queue factory.</param>
+        public ReactQueueConfigurationFactory(
+            IActionQueueFactory dispatcherQueueFactory,
+            IActionQueueFactory javaScriptQueueFactory,
+            IActionQueueFactory nativeModulesQueueFactory)
+        {
+            _dispatcherQueueFactory = dispatcherQueueFactory;
+            _javaScriptQueueFactory = javaScriptQueueFactory;
+            _nativeModulesQueueFactory = nativeModulesQueueFactory;
+        }
+
+        /// <summary>
+        /// The default queue configuration factory.
+        /// </summary>
+        public static ReactQueueConfigurationFactory Default
+        {
+            get
+            {
+                return new ReactQueueConfigurationFactory(
+                    new ActionQueueFactory(onError => new DispatcherActionQueue(onError, Dispatcher)),
+                    new ActionQueueFactory(onError => new ActionQueue(onError, NewThreadScheduler.Default)),
+                    new ActionQueueFactory(onError => new ActionQueue(onError)));
+            }
+        }
+
+#if WINDOWS_UWP
```

Comment:
can this be extracted out into a link-time substituted file that has a class extension in it? In ReactNative and ReactNative.Net46 it could be ReactQueueConfigurationFactory.Dispatcher.cs . if possible, I'd like to avoid adding new #if for UWP/WPF in new commits where its straightforward.

Reply:
Honestly, I think the next step will be to eliminate the DispatcherActionQueue in the ReactQueueConfiguration.  There's really no good reason to have it, since the Dispatcher implementations themselves are already FIFO and even the UWP Fall Creators Update has a built in DispatcherQueue implementation that could be useful. Let's go with this for now, in hopes that it will go away anyway, and refactor if necessary.